packages/core/src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -44,12 +52,70 @@ function setup(options: ReactNativeErrorHandlersOptions): void {
  * Setup unhandled promise rejection tracking
  */
 function setupUnhandledRejectionsTracking(patchGlobalPromise: boolean): void {
-  if (patchGlobalPromise) {
-    polyfillPromise();
-  }
+  try {
+    if (isHermesEnabled() && RN_GLOBAL_OBJ.HermesInternal?.enablePromiseRejectionTracker) {
```

Comment:
I think we should also check the `hasPromise` same as RN does it to ensure we are adding the tracker to the correct Promise.

https://github.com/facebook/react-native/blob/c89c5d7e3d981e5e11ae926bdd83567cbe61201a/packages/react-native/Libraries/Core/polyfillPromise.js#L25

Reply:
Good idea Krystof 👍 
Updated with 9fd5226f326a7bbd131c8dc7db77e4c06566e5ac